### PR TITLE
[WIP] webkitgtk: fix build on darwin

### PR DESCRIPTION
**Pull request by mjlbach**
###### Motivation for this change

This is not ready for review (but open to input).

WebkitGTK is currently broken on darwin, trying to continue on @matthewbauer's work and @jmercouris's patches from nyxt (which motivates this PR).

Right now, it seems like there are some apple specific checks not working, wondering if it's an 10.12 SDK issue:

```
❯ nix build '.#webkitgtk'
warning: Git tree '/Users/michael/Repositories/nix/nixpkgs' is dirty
error: builder for '/nix/store/ndcyvi13afsjvfdsvkrq0wqqr26z49lh-webkitgtk-2.30.5.drv' failed with exit code 1;
       last 10 log lines:
       > #if OS(DARWIN) && TARGET_OS_TV
       >                   ^
       > ../Source/WTF/wtf/PlatformOS.h:77:19: warning: 'TARGET_OS_WATCH' is not defined, evaluates to 0 [-Wundef]
       > #if OS(DARWIN) && TARGET_OS_WATCH
       >                   ^
       > ../Source/WTF/wtf/RAMSize.cpp:62:2: error: "Missing a platform specific way of determining the available RAM"
       > #error "Missing a platform specific way of determining the available RAM"
       >  ^
       > 3 warnings and 1 error generated.
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/ndcyvi13afsjvfdsvkrq0wqqr26z49lh-webkitgtk-2.30.5.drv'
```
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/webkitgtk/default.nix [hunk 1]
```diff
@@ -167,11 +196,16 @@ stdenv.mkDerivation rec {
...
-  ] ++ optional (stdenv.isLinux && enableGLES) "-DENABLE_GLES2=ON";
+    "-DUSE_SYSTEMD=OFF"
+    "-DUSE_APPLE_ICU=OFF"
+    "-DENABLE_WEB_CRYPTO=OFF"
+  ] ++ optional (stdenv.hostPlatform.isLinux && enableGLES) "-DENABLE_GLES2=ON";
 
   postPatch = ''
     patchShebangs .
-  '';
+  '' + ''
```
Comment (jtojnar) [inline pkgs/development/libraries/webkitgtk/default.nix:206]: Trailing whitespace.

File: pkgs/development/libraries/webkitgtk/default.nix [hunk 2]
```diff
@@ -63,20 +64,45 @@ stdenv.mkDerivation rec {
...
 
-  patches = optionals stdenv.isLinux [
+  patches = optionals stdenv.hostPlatform.isLinux [
     (substituteAll {
       src = ./fix-bubblewrap-paths.patch;
       inherit (builtins) storeDir;
     })
     ./libglvnd-headers.patch
-  ];
+    ./darwin/patch-WTF-wtf-Randomdevice.diff
```
Comment (jtojnar) [inline pkgs/development/libraries/webkitgtk/default.nix:80]: Could you add a comment to each patch explaining why it is needed, what it does and where does it come from? Ideally, we would use `fetchpatch` to get them.

Comment (mjlbach) [reply pkgs/development/libraries/webkitgtk/default.nix:80]: Yes, I will once I get it building. These patches are all taken from the MacPort of nyxt browser (https://source.atlas.engineer/view/repository/macports-port), however, the maintainer of the MacPort will not be continuing packaging it, so I put this in tree until we either get them upstream which the WebKitGTK maintainers are amenable to (these are for MacOS compatibility not nyxt) or find another home for them where we can keep them up to date.

**Comment by jtojnar** [comment]
Fixed by https://github.com/NixOS/nixpkgs/pull/123298?

**Comment by mjlbach** [comment]
Yep

**Pull request closed without merging**